### PR TITLE
chore(gatsby-remark-prismjs): add warnings when language not found

### PR DESCRIPTION
**Pull request by arturfsousa**
## Description

Adding a warning to the `gatsby-remark-prismjs` plugin when language is not found. The warn pops up only once per language to avoid spamming messages as @pieh commented in the issue #11667.

## Related Issues

Fixes: #11667

### CODE REVIEW

File: packages/gatsby-remark-prismjs/src/highlight-code.js [hunk 1]
```diff
@@ -3,6 +3,7 @@ const _ = require(`lodash`)
 
 const loadPrismLanguage = require(`./load-prism-language`)
 const handleDirectives = require(`./directives`)
+const notSupportedLanguages = []
```
Comment (DSchau) [inline packages/gatsby-remark-prismjs/src/highlight-code.js]: How about we use a Set here?

```suggestion
const unsupportedLanguages = new Set()
```

File: packages/gatsby-remark-prismjs/src/highlight-code.js [hunk 2]
```diff
@@ -14,6 +15,13 @@ module.exports = (language, code, lineNumbersHighlight = []) => {
       if (language === `none`) {
         return code // Don't escape if set to none.
       } else {
+        if (!notSupportedLanguages.includes(language)) {
```
Comment (DSchau) [inline packages/gatsby-remark-prismjs/src/highlight-code.js]:
```suggestion
        if (!unsupportedLanguages.has(language)) {
```

File: packages/gatsby-remark-prismjs/src/highlight-code.js [hunk 3]
```diff
@@ -14,6 +15,13 @@ module.exports = (language, code, lineNumbersHighlight = []) => {
...
+          console.warn(
+            `unable to find prism language '${language}' for highlighting.`,
+            `applying generic code block`
+          )
+          notSupportedLanguages.push(language)
```
Comment (DSchau) [inline packages/gatsby-remark-prismjs/src/highlight-code.js]:
```suggestion
          unsupportedLanguages.add(language)
```

Comment (DSchau) [reply packages/gatsby-remark-prismjs/src/highlight-code.js]: Also maybe we lowercase here? e.g. we wouldn't want to warn for JavaScript vs. javascript.

File: packages/gatsby-remark-prismjs/src/highlight-code.js (hunk 1 above)
Comment (arturfsousa) [reply packages/gatsby-remark-prismjs/src/highlight-code.js]: Nice catch!

File: packages/gatsby-remark-prismjs/src/highlight-code.js (hunk 3 above)
Comment (arturfsousa) [reply packages/gatsby-remark-prismjs/src/highlight-code.js]: Good idea. I will do it.

Comment (arturfsousa) [reply packages/gatsby-remark-prismjs/src/highlight-code.js]: Done @DSchau

**Pull request merged**